### PR TITLE
Make TaskEither a Monoid

### PR DESCRIPTION
**Pull request by mlegenhausen**
The added functions are a combination of the Monoid definitions from `Task` and `Either`.

### CODE REVIEW

File: src/TaskEither.ts [hunk 1]
```diff
@@ -174,6 +183,50 @@ export const fromPredicate = <L, A>(predicate: Predicate<A>, whenFalse: (a: A) =
...
+  return {
+    concat: (x, y) => new TaskEither<L, A>(S2.concat(x.value, y.value))
+  }
+}
+
+/**
+ * @function
+ * @since 1.9.0
+ */
+export const getMonoid = <L, A>(M: Monoid<A>): Monoid<TaskEither<L, A>> => {
```
Comment (gcanti) [inline src/TaskEither.ts]: This is unlawful, note that there's no `getMonoid` in the `Either` module.

Counterexample

```ts
it('right unit and a left', () => {
  return M.concat(left(delay(10, 'a')), M.empty)
    .run()
    .then(x => assert.deepEqual(x, eitherLeft('a'))) // fails
})
```

Comment (mlegenhausen) [reply src/TaskEither.ts]: Was already wondering. I will remove the `getMonoid` function.

File: test/TaskEither.ts [hunk 2]
```diff
@@ -215,31 +214,6 @@ describe('TaskEither', () => {
     })
   })
 
-  describe('getMonoid', () => {
```
Comment (gcanti) [inline test/TaskEither.ts]: we can keep the tests related to Semigroup though

Comment (mlegenhausen) [reply test/TaskEither.ts]: I added a separate set of `getSemigroup` tests which are equal to the tests from `Either`.

**Comment by gcanti** [comment]
Thanks @mlegenhausen

**Pull request merged**